src/use-input.ts

```diff
@@ -40,9 +40,11 @@ export const useInput = <T extends BaseInput>(host: T) => {
 
 		useEffect(() => {
 			const onMouseDown = <T extends Event>(e: T) => {
+				const element = e.target as HTMLSelectElement | null;
```

Comment:
We should check if the element itself is disabled not the event target.

Reply:
@megheaiulian you mean to implement something like:
`const element = document.querySelector('#input[part="input"]') | null;`
and in if statement check `element?.disabled` ?

![image](https://user-images.githubusercontent.com/111497107/200767874-cfc15bc4-a542-403f-8e6a-dc616f8779fd.png)